packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -287,7 +287,7 @@ async function getDeployedStack(cfn: aws.CloudFormation, stackName: string): Pro
 
 export async function readCurrentTemplate(cfn: aws.CloudFormation, stackName: string) {
   try {
-    const response = await cfn.getTemplate({ StackName: stackName, TemplateStage: 'Original' }).promise();
+    const response = await cfn.getTemplate({ StackName: stackName, TemplateStage: 'Processed' }).promise();
```

Comment:
So processed templates would compare a local stack against the expanded resources from transforms (macros).

i'm wondering if this should be an option in `cdk diff` rather than a default. Both modes of operation seem valid and useful